csodiaq/loaders/LibraryLoaderStrategyTable.py

```diff
@@ -0,0 +1,193 @@
+    intensities = df.groupby("csodiaqLibKey")["peakIntensity"].apply(list).to_dict()
+    df.drop_duplicates(subset="csodiaqLibKey", inplace=True)
+    df.set_index("csodiaqLibKey", drop=True, inplace=True)
+    df.drop(
+        ["precursorMz", "peakMz", "peptideName", "peakIntensity"], axis=1, inplace=True
+    )
+    metadata = df.to_dict(orient="index")
+    return {
+        "csodiaqKeys": keys,
+        "mz": mz,
+        "intensities": intensities,
+        "metadata": metadata,
+    }
+
+
+def _create_csodiaq_library_entry(
+    organizedDataDict: dict, maxPeakNum: int, csodiaqKeyIdx: int
+) -> dict:
+    csodiaqKey = organizedDataDict["csodiaqKeys"][csodiaqKeyIdx]
+    peaks = create_peaks_from_mz_intensity_lists_and_csodiaq_key_id(
+        organizedDataDict["mz"][csodiaqKey],
+        organizedDataDict["intensities"][csodiaqKey],
+        csodiaqKeyIdx,
+    )
+    reducedPeaks = remove_low_intensity_peaks_below_max_peak_num(peaks, maxPeakNum)
+    isDecoy = int(
+        "decoy" in organizedDataDict["metadata"][csodiaqKey]["proteinName"].lower()
+    )
+    return csodiaqKey, {
+        finalVariableNames["precursorCharge"]: organizedDataDict["metadata"][
+            csodiaqKey
+        ]["precursorCharge"],
+        finalVariableNames["identifier"]: organizedDataDict["metadata"][csodiaqKey][
+            "identifier"
+        ],
+        finalVariableNames["proteinName"]: organizedDataDict["metadata"][csodiaqKey][
+            "proteinName"
+        ],
+        finalVariableNames["peaks"]: sorted(reducedPeaks),
+        finalVariableNames["csodiaqKeyIdx"]: csodiaqKeyIdx,
+        finalVariableNames["isDecoy"]: isDecoy,
+    }
+
+
+def _set_old_to_new_column_dict(filePath) -> None:
+    librarySource = _determine_library_source_from_file(filePath)
+    newColumns = [
+        "precursorMz",
+        "peptideName",
+        "peakMz",
+        "peakIntensity",
+        "precursorCharge",
+        "identifier",
+        "proteinName",
+    ]
+    if librarySource == "spectrast":
+        oldColumns = [
+            "PrecursorMz",
+            "FullUniModPeptideName",
+            "ProductMz",
+            "LibraryIntensity",
+            "PrecursorCharge",
+            "transition_group_id",
+            "ProteinName",
+        ]
+        return dict(zip(oldColumns, newColumns))
+    elif librarySource == "fragpipe":
+        oldColumns = [
+            "PrecursorMz",
+            "ModifiedPeptideSequence",
+            "ProductMz",
+            "LibraryIntensity",
+            "PrecursorCharge",
+            "PeptideSequence",
+            "ProteinId",
+        ]
+        return dict(zip(oldColumns, newColumns))
+    elif librarySource == "prosit":
+        oldColumns = [
+            "PrecursorMz",
+            "ModifiedPeptide",
+            "FragmentMz",
+            "RelativeIntensity",
+            "PrecursorCharge",
+            "StrippedPeptide",
+            "FragmentLossType",
+        ]
```

Comment:
The library formats are hard-coded 100+ lines into the code; updating/modifying this will be more difficult.
As a patch, I might recommend moving these to the top of the file (global variables - not ideal). Having this as a separate file that contains the mapping might be preferable, e.g.:  
`if librarySource == "something":`  
`    from csodiaq.loaders.mappings import something_mapper`  
`    return something_mapper`

Reply:
Great idea. Mapped in a separate file now